src/EditorFeatures/Core/Tagging/AbstractAsynchronousTaggerProvider.TagSource_ProduceTags.cs

```diff
@@ -514,7 +540,7 @@ private Task ProduceTagsAsync(TaggerContext<TTag> context, CancellationToken can
 
                 foreach (var (latestBuffer, latestSpans) in newTagTrees)
                 {
-                    var snapshot = spansToTag.First(s => s.SnapshotSpan.Snapshot.TextBuffer == latestBuffer).SnapshotSpan.Snapshot;
+                    var snapshot = GetSpanForBuffer(spansToTag, latestBuffer).SnapshotSpan.Snapshot;
```

Comment:
>GetSpanForBuffer [](http://example.com/codeflow?start=35&length=16)

alternatively could use FirstOrDefault

Reply:
Didn't seem to exist on OneOrMany. But I can check again!

Reply:
done!